CMake/ClangCompilers.cmake

```diff
@@ -54,9 +54,8 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror=vla")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wvla")
 
 # set compiler warnings
-set(CMAKE_CXX_FLAGS
-    "${CMAKE_CXX_FLAGS} -Wall -Wno-unused-variable -Wno-overloaded-virtual -Wno-unused-private-field -Wno-unused-local-typedef"
-)
+string(APPEND CMAKE_CXX_FLAGS " -Wall -Wno-unused-variable -Wno-overloaded-virtual -Wno-unused-private-field -Wno-unused-local-typedef -Wsuggest-override")
```

Comment:
Seems like `-Wsuggest-override` is recognized by GNU not by clang, thus creating several warnings for the latter. Which clang-10 I get: `warning: unknown warning option '-Wsuggest-override'; did you mean '-Wshift-overflow'? [-Wunknown-warning-option]`

Reply:
@williamfgc Oops, looks like -Wsuggest-override was added in clang-11
https://reviews.llvm.org/D82728
https://clang.llvm.org/docs/DiagnosticsReference.html#wsuggest-override

Reply:
#3274 should fix this